django/conftest.py

```diff
@@ -78,6 +83,38 @@ def http_server():
     thread.join()
 
 
+class KeyProviderGetHTTPRequestHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        keys = json.dumps(
+            {
+                "public_keys": [
+                    {
+                        "key_identifier": "FETCHED_KEY_IDENTIFIER",
+                        "key": "FETCHED_TEST_KEY",
+                        "is_current": True,
+                    }
+                ]
+            }
+        ).encode()
+        self.send_response(200)
+        self.send_header("Content-Type", "application/json")
+        self.end_headers()
+        self.wfile.write(keys)
+        return
+
+
+@pytest.fixture()
+def http_server_for_provider_keys():
```

Comment:
Honestly, I made it this way because I didn't want to use requests-mock. For some unknown reason, I'm hesitant on adding libraries to the project. But sure I can use it if we are fine with adding the library to the project 👍

Reply:
Also emulating a full server seemed more straight forward :D